### ISSUE TITLE
Consider calling ConfigureAwait on the awaited task!!

### ISSUE THREAD

**Issue by AhmedMKamal** [dotnet/roslyn-analyzers#3034]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.7 (Latest)

#### Diagnostic ID

Example: [CA2007](https://docs.microsoft.com/en-gb/visualstudio/code-quality/ca2007?view=vs-2019)

#### Repro steps

```c#
// awaiting task in MVC action.
public async Task<IActionResult> Index()
{
        return View(await _someService.LoadDataFromEFCoreContextAsync());
}
```

#### Expected behavior
the analyzer shouldn't care about that task since in `.net core` it doesn't matter if you call `ConfigureAwait` on the awaited task or not **AFAIK**.

#### Actual behavior
I'm getting this warning `warning CA2007: Consider calling ConfigureAwait on the awaited task`

**Comment by mavasani** [comment]
This is by design. The intent of this rule is that all async calls have an explicit ConfigureAwait for _clarity_. Please see the documentation here: https://docs.microsoft.com/en-gb/visualstudio/code-quality/ca2007?view=vs-2019#how-to-fix-violations

```
Calling ConfigureAwait(true) on the task has the same behavior as not explicitly calling ConfigureAwait. By explicitly calling this method, you're letting readers know you intentionally want to perform the continuation on the original synchronization context.

You can suppress this warning if you know that the consumer is not a graphical user interface (GUI) app or if the consumer does not have a SynchronizationContext.
```
